.github/workflows/lint.yml

```diff
@@ -0,0 +1,25 @@
+name: lint
+
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
```

Comment:
There's some subtlety here worth pointing out.  `push` means run the check on any commit that gets added to the specified branch (`main` in this case), so any time `main` gets updated, either via a PR or a direct push, it will run the check.  `pull_request` runs the check in the hypothetical result of merging the PR into the target branch (which in general is different from the current state of the PR branch).  If `push` were not restricted to `main`, then any time you open a PR, both `push` and `pull_request` would run, which I don't think is useful.